tests/helper/test_helper.py

```diff
@@ -604,14 +608,14 @@ def test_process_media_empty_url(
 @pytest.mark.parametrize(
     "returncode, error",
     [
-        (0, []),
+        (1 if PLATFORM == "WSL" else 0, []),
         (
-            1,
+            0 if PLATFORM == "WSL" else 1,
             [
                 " The tool ",
                 ("footer_contrast", "xdg-open"),
                 " did not run successfully" ". Exited with ",
-                ("footer_contrast", "1"),
+                ("footer_contrast", "0" if PLATFORM == "WSL" else "1"),
             ],
         ),
     ],
```

Comment:
This test didn't consider the different return codes in WSL, as given in `platform_code.py` line 92.

```python
def successful_GUI_return_code() -> int:  # noqa: N802 (allow upper case)
    """
    Returns success return code for GUI commands, which are OS specific.
    """
    # WSL uses GUI return code as 1. Refer below link to know more:
    # https://stackoverflow.com/questions/52423031/
    # why-does-opening-an-explorer-window-and-selecting-a-file-through-pythons-subpro/
    # 52423798#52423798
    if PLATFORM == "WSL":
        return 1

    return 0```

Reply:
`explorer.exe` seems to always return a non-zero exit code, regardless of whether the task is successful.

More information could be found on this [issue](https://github.com/microsoft/WSL/issues/6565).

Reply:
> we can refactor the function you quoted, such that it returns some value, depending upon a passed-in return code. This would move the comparison, and hence these explicit error codes, into a platform-dependent location.

Implemented this approach, thanks for the suggestion! Also will look into refactoring some `open_media` logic to a platform-specific location in another PR.